pennylane/vqe/vqe.py

```diff
@@ -552,6 +552,10 @@ def queue(self, context=qml.QueuingContext):
         context.append(self, owns=tuple(self.ops))
         return self
 
+    def diagonalizing_gates(self):
+        # todo: find better solution for observables whose diagonalisation we cannot compute
+        return []
```

Comment:
I feel like `raise NotImplementedError` was a nice solution, shame that the specs forces us to define it!

Reply:
Yes, I also prefer the error, because then we can catch other problems. Ok, let me do the try-except block!